GitUI/BranchTreePanel/StashTree.cs

```diff
@@ -0,0 +1,95 @@
+﻿using GitCommands;
+using GitUI.UserControls.RevisionGrid;
+using GitUIPluginInterfaces;
+using Microsoft.VisualStudio.Threading;
+
+namespace GitUI.BranchTreePanel
+{
+    internal sealed class StashTree : BaseRevisionTree
+    {
+        public StashTree(TreeNode treeNode, IGitUICommandsSource uiCommands, ICheckRefs refsSource)
+            : base(treeNode, uiCommands, refsSource)
+        {
+        }
+
+        internal void Refresh(Lazy<IReadOnlyCollection<GitRevision>> getStashRevs)
+        {
+            if (!IsAttached)
+            {
+                return;
+            }
+
+            ThreadHelper.JoinableTaskFactory.RunAsync(async () =>
+            {
+                await ReloadNodesAsync((token, _) =>
+                {
+                    Task<Nodes>? loadNodesTask = null;
+                    _updateSemaphore.WaitAsync();
+                    try
+                    {
+                        loadNodesTask = LoadNodesAsync(token, getStashRevs);
+                    }
+                    finally
+                    {
+                        _updateSemaphore.Release();
+                    }
+
+                    return loadNodesTask;
+                }, null).ConfigureAwait(false);
+            });
```

Comment:
❔ Do it like in BaseRevisionTree?
```
    <snip>
}).FileAndForget();
```
---

General question: Might `ConfigureAwait()` be the cause why the app sometimes crashes without any popup?

Reply:
Same for Branches/Tags in BaseRefTree.cs, so not changed.
I guess it could be changed.

> General question: Might ConfigureAwait() be the cause why the app sometimes crashes without any popup?

Maybe Debug.Assert or Debug.Fail that fires if you run in Debug. Hard to debug. Maybe fixed in .net7
https://github.com/dotnet/winforms/issues/3739